bot/exts/moderation/dm_relay.py

```diff
@@ -1,132 +1,68 @@
-
-        try:
-            self.webhook = await self.bot.fetch_webhook(self.webhook_id)
-        except discord.HTTPException:
-            log.exception(f"Failed to fetch webhook with id `{self.webhook_id}`")
-
-    @Cog.listener()
-    async def on_message(self, message: discord.Message) -> None:
-        """Relays the message's content and attachments to the dm_log channel."""
-        # Only relay DMs from humans
-        if message.author.bot or message.guild or self.webhook is None:
+        output = ""
+        async for msg in user.history(limit=limit, oldest_first=True):
+            created_at = msg.created_at.strftime(r"%Y-%m-%d %H:%M")
+
+            # Metadata (author, created_at, id)
+            output += f"{msg.author} [{created_at}] ({msg.id}): "
+
+            # Content
+            if msg.content:
+                output += msg.content + "\n"
+
+            # Embeds
+            if (embeds := len(msg.embeds)) > 0:
+                output += f"<{embeds} embed{'s' if embeds > 1 else ''}>\n"
+
+            # Attachments
+            attachments = "\n".join(a.url for a in msg.attachments)
+            if attachments:
+                output += attachments + "\n"
+
+        if not output:
+            await ctx.send(f"{Emojis.cross_mark} No direct message history with {user.mention}.")
             return
 
-        if message.clean_content:
-            await send_webhook(
-                webhook=self.webhook,
-                content=message.clean_content,
-                username=f"{message.author.display_name} ({message.author.id})",
-                avatar_url=message.author.avatar_url
-            )
-            await self.dm_cache.set("last_user", message.author.id)
-            self.bot.stats.incr("dm_relay.dm_received")
-
-        # Handle any attachments
-        if message.attachments:
-            try:
-                await send_attachments(
-                    message,
-                    self.webhook,
-                    username=f"{message.author.display_name} ({message.author.id})"
-                )
-            except (discord.errors.Forbidden, discord.errors.NotFound):
-                e = discord.Embed(
-                    description=":x: **This message contained an attachment, but it could not be retrieved**",
-                    color=Color.red()
-                )
-                await send_webhook(
-                    webhook=self.webhook,
-                    embed=e,
-                    username=f"{message.author.display_name} ({message.author.id})",
-                    avatar_url=message.author.avatar_url
-                )
-            except discord.HTTPException:
-                log.exception("Failed to send an attachment to the webhook")
-
-    async def cog_check(self, ctx: commands.Context) -> bool:
+        metadata = textwrap.dedent(f"""\
+            User: {user} ({user.id})
+            Channel ID: {user.dm_channel.id}\n
+        """)
```

Comment:
```suggestion
        metadata = (
            f"User: {user} ({user.id})\n"
            f"Channel ID: {user.dm_channel.id}\n\n"
        )
```
Using implicit line continuation and joining looks a bit nicer to me; the textwrap import can also be removed if you decide to use this